lib/client.js

```diff
@@ -597,7 +600,17 @@ MqttClient.prototype._cleanUp = function (forced, done) {
  */
 MqttClient.prototype._sendPacket = function (packet, cb) {
   if (!this.connected) {
-    return this.queue.push({ packet: packet, cb: cb });
+    if (0 === packet.qos && this.qosZeroNotQueued) {
```

Comment:
you should also check if `cmd`  is `publish`. The `qos` bit has different meaning with other packets.